app/services/description_import.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class DescriptionImport
+  include Dry::Monads[:result]
+
+  def self.import(description:, csv:)
+    new(description: description, csv: csv).import
+  end
+
+  def initialize(description:, csv:)
+    @description = description
+    @csv = csv
+  end
+
+  def import
+    params = @description.to_h
+
+    row = @csv.first
+    # The source_id is only there for the user to reference and should be ignored for data processing
+    headers = row.headers.excluding('source_id')
+
+    headers.each do |address|
+      split_address = address.scan(/[[:alpha:]]+|[[:digit:]]+/)
+                             .map { |item| /\d+/.match?(item) ? item.to_i - 1 : item.to_sym }
+      visit(params, split_address, row[address])
+    end
+
+    Success(Cocina::Models::Description.new(params))
+  rescue Cocina::Models::ValidationError => e
+    Failure(e.message)
+  end
+
+  private
+
+  def nest_hashes(value, *keys)
+    return value if keys.empty?
+
+    key = keys.shift
+    val = keys.empty? ? value : nest_hashes(value, *keys)
+    key.is_a?(Integer) ? [].tap { |arr| arr[key] = val } : { key => val }
+  end
+
+  def visit(what, rest, value, path = [])
```

Comment:
I'm unclear on what `what` and `rest` would be. Are those variable naming conventions we use in SDR?

Reply:
In this case `what` is a tree like data structure.  It's one of the nodes in the Cocina descriptive.  `rest` is the "path" we have to follow to get to the location in the tree.

Reply:
I'm not sure I would understand that from these names--aybe a comment would help, if there aren't other names for these variables that would convey that.

Reply:
I'm not great with names, do you have suggestions that would make it clearer?

Reply:
I've added yard doc to document the method parameters.

Reply:
Thanks, that helps. I was going to suggest `loc` or `location` for `rest` and `tree` for `what`.